app/workers/kyc/kycaid/address_worker.rb

```diff
@@ -0,0 +1,50 @@
+require 'sidekiq'
+
+module KYC
+  module Kycaid
+    class AddressWorker
+      include Sidekiq::Worker
+
+      def perform(params = {})
+        @params = params.symbolize_keys
+        
+        @user = User.find(@params[:user_id])
+        @applicant_id = @user.profiles.last.applicant_id
+        @document = Document.find_by(identificator: @params[:identificator])
+        address = ::KYCAID::Address.create(address_params)
+
+        if address.error || address.errors
+          Rails.logger.info("Error in document creation for: #{@user.uid}: #{address.errors} #{address.error}")
+          @user.labels.find_by(key: :address, scope: :private).update(value: 'rejected')
+        elsif address.document_id
+          @document.update(metadata: { address_id: address.document_id }.to_json)
+          verification = ::KYCAID::Verification.create(verification_params)
+        end
+      end
+
+      def address_params
+        {
+          front_file: {
+            tempfile: URI.open(@document.upload.url),
+            file_extension: @document.upload.file.extension,
+            file_name: @document.upload.file.filename,
+          },
+          type: 'ADDRESS_DOCUMENT',
```

Comment:
Yes, type in this case should be REGISTERED or BUSINESS or ADDITIONAL. In our case its ok to hardcode REGISTERED.
About ADDRESS_DOCUMENT - its already applied in gem https://github.com/openware/kycaid-ruby/blob/master/lib/kycaid/address.rb#L20